src/main/java/com/minecolonies/coremod/entity/pathfinding/WalkToProxy.java

```diff
@@ -201,7 +233,21 @@ private BlockPos getMinerProxy(final BlockPos target, final double distanceToPat
             //Check if miner is underground in shaft and his target is overground.
             if (workerY <= levelDepth && targetY > levelDepth)
             {
-                proxyList.add(new BlockPos(ladderPos.getX(), level.getDepth(), ladderPos.getZ()));
+                if(level.getRandomNode() != null && level.getRandomNode().getParent() != null)
+                {
+                    com.minecolonies.coremod.entity.ai.citizen.miner.Node currentNode = level.getNode(level.getRandomNode().getParent());
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Replace this fully qualified name with "Node" [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1942)